Astro 5.12 blog (#1691)
withastro/astro.build: src/content/blog/astro-5120.mdx

	}),
});
export const collections = { spacecraft };
```

You can then access the TOML content in your Astro components:

```astro title="src/pages/spacecraft.astro"
---
import { getCollection } from 'astro:content';

const spacecraft = await getCollection('spacecraft');
---

<ul>
	{
		spacecraft.map((item) => (
			<li>
				<h2>{item.name}</h2>
				<p>Launch Date: {item.launch_date.toLocaleDateString()}</p>
				<p>Crew: {item.crew.join(', ')}</p>
			</li>
		))
	}
</ul>
```

## New Netlify dev experience

The [Netlify adapter](https://docs.astro.build/en/guides/integrations-guide/netlify/) is now powered by Netlify's very own [Vite plugin](https://docs.netlify.com/frameworks/vite/#vite-plugin) during development, bringing Netlify's entire platform and primitives to your local dev environment!

### The power of Netlify on localhost

Until now, using Netlify primitives with Astro during local development required running your project through the [Netlify CLI](https://docs.netlify.com/cli/get-started/). The new adapter embeds these capabilities directly into Astro's dev server instead.

After upgrading to the latest version of the adapter, running `astro dev` will provide:

> - A **local Netlify Image CDN** for [Astro images](https://docs.astro.build/en/guides/images/)
> - A **local Blobs server** for [sessions](https://docs.astro.build/en/guides/sessions/) 
> - **[Redirects, rewrites](https://docs.netlify.com/routing/redirects/) and [headers](https://docs.netlify.com/routing/headers/#app)**  from your Netlify config
> - The **[Netlify Edge Context](https://docs.astro.build/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site)** in on-demand pages

Reviewer: ```suggestion
- A local Netlify Image CDN for [Astro images](https://docs.astro.build/en/guides/images/)
- A local Blobs server for [sessions](https://docs.astro.build/en/guides/sessions/) 
- Redirects, rewrites](https://docs.netlify.com/routing/redirects/) and [headers](https://docs.netlify.com/routing/headers/#app)  from your Netlify config
- The [Netlify Edge Context](https://docs.astro.build/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site) in on-demand pages
```
The bold really didn't work visually in the blog post, so I think we can just go with the list! (The non-bold "A" in front became really difficult to read in comparison, one line had bold links but the others didn't...)

> ### Environment variables

Reviewer: This feels weird to be one item that should be in the above list, plus we're only mentioning the actual new Astro config option in passing.

I'm not 100% sure yet what the solution should look like, but I think it makes more sense to have a flow something like:

```
## New Netlify dev experience

- until now, CLI
- update adapter for new experience, with config option to customize some features
- list (including environment variables as an list item

### devFeatures

- Along with new dev experience comes a way to customize it
- `devFeatures.images` and `devFeatures.environmentVariables` each described, both mentioning their corresponding Netlify feature advantages (how they make your experience better in dev), their defaults (and why), and showing configuring non-defaults.

These features are a zero-config upgrade, requiring no changes to your project.

Additionally, you can configure the optional `devFeatures` object to automatically populate your development server with the [environment variables](https://docs.netlify.com/environment-variables/overview/) from your [linked](https://docs.netlify.com/cli/get-started/#link-and-unlink-sites) Netlify site:

```js ins={7}
import { defineConfig } from 'astro/config'
import netlify from '@astrojs/netlify'

export default defineConfig({
  adapter: netlify({
    devFeatures: {
      environmentVariables: true,
    },
  })
})
```

Now you can enjoy the rich, Netlify flavor in your own home!

<cite>Thanks to [Eduardo Bouças](https://github.com/eduardoboucas) and [chee rabbits](https://github.com/chee) from Netlify who contributed this feature. Extra thanks to chee who contributed to this post too.</cite>
## Experimental raw environment values

> A new experimental configuration option allows you to opt out of Astro's automatic type coercion for environment variable values in `import.meta.env`. By default, Astro converts string values like `"true"`, `"false"`, `"1"`, and `"0"` to their respective types, but this can be unexpected since environment variables are generally expected to be strings. This was not intended behavior, and was the result of a feature that was removed in Astro 5.0. Removing this for everyone would be a breaking change, so we are introducing an experimental option to opt out of this behavior. This will probably become the default in a future release.

Reviewer: ```suggestion
A new experimental configuration option allows you to opt out of Astro's automatic type coercion for environment variable values in `import.meta.env`. By default, Astro converts string values like `"true"`, `"false"`, `"1"`, and `"0"` to their respective types, but this can be unexpected since environment variables are generally expected to be strings.

This aligns Astro's behavior with [Vite's handling of environment variables](https://vite.dev/guide/env-and-mode.html#env-variables), and is likely to become the default in a future major release.
```
I don't think any of the backstory is interesting/provides any value here? But the key thing we've said in the changeset and docs is that this aligns behaviour with Vite, so that makes it sound more legit, and feels like a real reason to do this.

> Enable this feature in your Astro configuration:

Reviewer: ```suggestion
To enable this feature, add the `rawEnvValues` flag to your Astro configuration:
```

```js title="astro.config.mjs" ins={3}
export default defineConfig({
	experimental: {
		rawEnvValues: true,
	},
});
```

> This experimental option only affects the `import.meta.env` object and doesn't impact the `astro:env` module.

Reviewer: ```suggestion
This experimental option only affects the `import.meta.env` object and doesn't impact the `astro:env` module that powers Astro's [type-safe environment variables](https://docs.astro.build/en/guides/environment-variables/#type-safe-environment-variables).
```

- A local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/#local-development) server. This is used for [images](#netlify-image-cdn-support) by default
- A local [Netlify Blobs](https://docs.netlify.com/build/data-and-storage/netlify-blobs/) server. This is used for [sessions](#sessions) by default
- [Redirects, rewrites](https://docs.netlify.com/routing/redirects/) and [headers](https://docs.netlify.com/routing/headers/#app) from your Netlify config
- The [Netlify Edge Context](https://docs.astro.build/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site) in on-demand pages
- [Environment variables](https://docs.netlify.com/environment-variables/overview/) populated from your linked Netlify site

### Configuring dev features

Along with these features comes a way to configure them. The Netlify adapter now accepts a `devFeatures` option in its configuration, allowing you to enable or disable specific features during development. You can set this to `true` or `false` to enable or disable all features, or you can configure them individually. Currently the features you can configure are:

> - `images`: Enables the local Netlify Image CDN for Astro images during dev. Default to `true`.

Reviewer: ```suggestion
- `images`: Enables the local Netlify Image CDN for Astro images during dev. Defaults to `true`.
```

  adapter: netlify({
    devFeatures: {
      environmentVariables: true,
      images: false,
    },
  })
})
```

Now you can enjoy the rich Netlify flavor in your own home!

<cite>Thanks to [Eduardo Bouças](https://github.com/eduardoboucas) and [chee rabbits](https://github.com/chee) from Netlify who contributed this feature. Extra thanks to chee who contributed to this post too.</cite>
## Experimental raw environment values

A new experimental configuration option allows you to opt out of Astro's automatic type coercion for environment variable values in `import.meta.env`. By default, Astro converts string values like `"true"`, `"false"`, `"1"`, and `"0"` to their respective types, but this can be unexpected since environment variables are generally expected to be strings.

This aligns Astro's behavior with [Vite's handling of environment variables](https://vite.dev/guide/env-and-mode.html#env-variables), and is likely to become the default in a future major release.

To enable this feature, add the `rawEnvValues` flag to your Astro configuration:

```js title="astro.config.mjs" ins={3}
export default defineConfig({
	experimental: {
		rawEnvValues: true,
	},
});
```

> This experimental option only affects the `import.meta.env` object and doesn't impact the `astro:env` module that powers Astro's [type-safe environment variables](https://docs.astro.build/en/guides/environment-variables/#type-safe-environment-variables).

Reviewer: ```suggestion
This experimental option only affects the `import.meta.env` object and doesn't impact the `astro:env` module that powers Astro's [type-safe environment variables](https://docs.astro.build/en/guides/environment-variables/#type-safe-environment-variables).

Thanks to [Adam Langbert](https://github.com/adamhl8) for contributing this feature!
```

> 

Reviewer: lol, at first I thought you meant "of `astro:env`", which would be YOU! 😂

After upgrading to the latest version of the adapter, running `astro dev` will provide:

- A local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/#local-development) server. This is used for [images](#netlify-image-cdn-support) by default
- A local [Netlify Blobs](https://docs.netlify.com/build/data-and-storage/netlify-blobs/) server. This is used for [sessions](#sessions) by default
- [Redirects, rewrites](https://docs.netlify.com/routing/redirects/) and [headers](https://docs.netlify.com/routing/headers/#app) from your Netlify config
- The [Netlify Edge Context](https://docs.astro.build/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site) in on-demand pages
- [Environment variables](https://docs.netlify.com/environment-variables/overview/) populated from your linked Netlify site

### Configuring dev features

> Along with these features comes a way to configure them. The Netlify adapter now accepts a `devFeatures` option in its configuration, allowing you to enable or disable specific features during development. You can set this to `true` or `false` to enable or disable all features, or you can configure them individually. Currently the features you can configure are:

Reviewer 2: comma nit
```suggestion
Along with these features comes a way to configure them. The Netlify adapter now accepts a `devFeatures` option in its configuration, allowing you to enable or disable specific features during development. You can set this to `true` or `false` to enable or disable all features, or you can configure them individually. Currently, the features you can configure are:
```

> Along with these features comes a way to configure them. The Netlify adapter now accepts a `devFeatures` option in its configuration, allowing you to enable or disable specific features during development. You can set this to `true` or `false` to enable or disable all features, or you can configure them individually. Currently the features you can configure are:

Reviewer 2: On a more higher level, we use "enable and disable" two times here and kind of repeat the information, maybe using "toggle" and trimming it a bit could work better? Not sure if the way I wrote makes it clear the difference between passing an object for specific features (example below) or all features (passing a boolean directly to `devFeatures`)
```suggestion
Along with these features comes a way to configure them. The Netlify adapter now accepts a `devFeatures` option in its configuration, allowing you to toggle specific or all features (passing a boolean directly) during development. Currently, the features you can configure are:
```

})
```

> Now you can enjoy the rich, Netlify flavor in your own home!

Reviewer 3: 🩵
